src/Symfony/Component/String/AbstractUnicodeString.php

```diff
@@ -220,6 +220,18 @@ public function lower(): static
         return $str;
     }
 
+    public function localeLower(string $locale): static
+    {
+        if (null !== $transliterator = $this->getLocaleTransliterator($locale, 'Lower')) {
+            $str = clone $this;
+            $str->string = $transliterator->transliterate($str->string);
+
+            return $str;
+        }
+
+        return $this->lower();
```

Comment:
I wonder if this silent fallback is the right thing to do. If someone calls `localeLower()`, they explicitly opted into using a locale-aware transliterator. If that transliterator is not available (e.g. because of a typo in the passed locale), how would they be made aware of that?

Reply:
Not all locales have special rules, the fallback is the correct behavior for them.

Reply:
I see. So we cannot tell apart a mistyped locate from a valid one without a special transliterator?

Reply:
This is the same behaviour as String.toLocaleUpperCase().

const city = 'istanbul';
console.log(city.toLocaleUpperCase('foobar')); // "ISTANBUL"

Reply:
The transliterator use IDs composed with source and filters

> The source specifier describes the characters or strings that the transform will modify. The target specifier describes the result of the modification. If the source is not given, then the source is “Any”, the set of all characters. Source and Target specifiers can be [Script IDs](http://www.unicode.org/cldr/utility/properties.jsp#Script) (long like “Latin” or short like “Latn”), [Unicode language Identifiers](http://www.unicode.org/reports/tr35/#Unicode_Language_and_Locale_Identifiers) (like fr, en_US, or zh_Hant), or special tags (like Any or Hex)

https://unicode-org.github.io/icu/userguide/transforms/general/#basic-ids 

We could validate the identifier/locale passed in parameter (with [locale_parse](https://www.php.net/manual/fr/locale.parselocale.php) ?) but indeed, nothing guarantee the matching transliterator does exist.

Reply:
All right, makes sense. Let's keep it this way. Thank you for the explanation!